src/AlgebraResultTypes.jl

```diff
@@ -75,4 +76,36 @@ function result_ring(::Type{T}, ::Type{S}, Ts...) where {T <: Number, S <: Numbe
     result_ring(result_ring(T, S), Ts...)
 end
 
+function result_group(op,::Type{T}) where {T <: Number}
+    o = one(T)
+    typeof(op(o,inv(o)))
```

Comment:
the definition of group requires an inverse operation to be defined, so `inv` is needed for closure

Reply:
But the operation here is generic (eg `+`), and `inv` is `x -> 1/x`.

Also, the point of this package is just to figure out the type of `op(::A, ::B)` from `A` and `B`. `op(o, o)` should suffice.

Reply:
@tpapp It would be then misleading to call this a group.  I wonder if it's useful to have `result_monoid` (no inverse) or `result_semigroup` (also no identity).

Also, using `one` here is not correct for the same reason (this does not produce an identity element for a generic operation).  I don't know if there is a public API that does this but I know that `reduce` uses `Base.reduce_empty` internally.  Maybe you can use it or take some code from it.

Reply:
the main issue is that `inv` gives a multiplicative inverse and not an additive inverse. unfortunately, Julia does not have any syntax which automatically tells you what inverse method to use, maybe it should be provided as an input to the function `result_group(op, inv, ::Type)`, then it can be called with the correct inverse methods, such as `-` for additive and `inv` for multiplicative.

Reply:
@tkf: I guess `Base.reduce_empty` is another approach to this problem, also for a special case.

Having an identity or not is an orthogonal problem. Eg fields also have multiplicative and additive identities, but this package just does a type calculation, the requirements of which are eg [documented here](https://github.com/tpapp/AlgebraResultTypes.jl/blob/446a3baa7fc0766d570883dc57e4318e2246ddd6/src/AlgebraResultTypes.jl#L29-L36).